### PR TITLE
Do not reset cursor position during test discovery

### PR DESCRIPTION
**Pull request by sharwell**
😑

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/ModuleInitializer.cs [hunk 1]
```diff
@@ -16,12 +16,6 @@ public static void InitializeModule()
             // handling of a Windows message.
             AppContext.SetSwitch("System.Windows.Forms.DoNotCatchUnhandledExceptions", isEnabled: true);
 
-            // Some controls have behavior that changes when the mouse is over them. Make sure we start with the cursor
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/UnitTests/ModuleInitializer.cs:19]: @sharwell, can you add few details on what you observed with this and how it was interfering with tests?

**Comment by dreddy-work** [comment]
@JeremyKuhne as FYI.

File: src/System.Windows.Forms/tests/UnitTests/ModuleInitializer.cs (hunk 1 above)
Comment (sharwell) [reply src/System.Windows.Forms/tests/UnitTests/ModuleInitializer.cs:19]: The mouse randomly moved while editing in winforms solution.

**Comment by kirsan31** [comment]
Currently the change in `DataGridViewTests.cs` is not necessary. I'll try to help a little with what's going on here. 

- Originally cursor position was changed in [tests/TestUtilities/ThreadExceptionFixture.cs](https://github.com/dotnet/winforms/pull/7031/files) for flaky DataGridView tests. But this attempt was no success (because UTs are run in parallel).
- Then https://github.com/dotnet/winforms/issues/7799 was created. _And due to uncontrollable circumstances, I dropped out of this process for a long time :(_
- In https://github.com/dotnet/winforms/pull/8574 these tests were removed but cursor position in `ThreadExceptionFixture.cs` was missed :(
- @JeremyKuhne removed `ThreadExceptionFixture.cs` in https://github.com/dotnet/winforms/pull/8695 and transfer it's logic to `UnitTests/ModuleInitializer.cs` with some modification.

For now we have no cursor dependent tests in `DataGridViewTests.cs` so no need to set cursor position there.

> The Cursor tests were the ones modifying the cursor position and causing intermittent errors in other tests. Moved them to their own collection and forced them to run sequentially.

This is very cool, because similar will be needed if we try to fix https://github.com/dotnet/winforms/issues/8535

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTests.cs [hunk 2]
```diff
@@ -11,6 +11,15 @@ namespace System.Windows.Forms.Tests
 {
     public partial class DataGridViewTests
     {
+        public DataGridViewTests()
+        {
+            // Some controls have behavior that changes when the mouse is over them. Make sure we start with the cursor
+            // at a known position.
+            //
+            // See https://github.com/dotnet/winforms/pull/7031#issuecomment-1101339968 for an example of this.
+            Cursor.Position = default;
```
Comment (kirsan31) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTests.cs:20]: I think that this change is not necessary. Pls see my [comment](https://github.com/dotnet/winforms/pull/8927#issuecomment-1492893171) in conversation.

**Comment by sharwell** [comment]
@kirsan31 I'm probably not the best one to comment on the need for this. For me, the important part is just the removal from the module initializer. As long as the mouse stops jumping around randomly while I'm trying to work, the rest can be anything. It's not clear if you want the removal of this call to be a follow-up, or if you want me to combine the fixing of the module initializer with the separate discussion of the need for cursor controls.

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by kirsan31** [comment]
@sharwell 

> I'm probably not the best one to comment on the need for this. For me, the important part is just the removal from the module initializer. As long as the mouse stops jumping around randomly while I'm trying to work, the rest can be anything.

+1
> It's not clear if you want the removal of this call to be a follow-up, or if you want me to combine the fixing of the module initializer with the separate discussion of the need for cursor controls.

I just wanted to clarify how and why original code (in ThreadExceptionFixture.cs) appeared.
And point out that it should have been removed in https://github.com/dotnet/winforms/pull/8574 (this pr was before JeremyKuhne's one).
If this code had remained unchanged until now, I would definitely say that it should 100% be removed.  But @JeremyKuhne  modified it, so now he has to make a decision based on what I wrote above.

**Review by KlausLoeffelmann — APPROVED** [review]

**Comment by JeremyKuhne** [comment]
@kirsan31 as there is no guarantee where the mouse will be when starting to run tests I'd love to get this back into a pre-run setup. I don't know how plausible this sort of thing is utilizing xunit functionality. I also don't know if it is plausible to detect that we're in discovery mode in the module initializer.

Personally I like the idea of adding analyzers to the unit tests that flag when we're doing global state things like repositioning the mouse as these sorts of things can be hard to find from intermittently failing tests.

If only there were more time...

**Comment by sharwell** [comment]
@JeremyKuhne My recommendation for that case is to detect the presence of CI execution via an environment variable, and not move the cursor at all for non-CI scenarios.

**Pull request merged**